src/ast/utils.rs

```diff
@@ -50,49 +51,50 @@ impl FreeKindedIdents {
 impl Visit for FreeKindedIdents {
     fn visit_nat(&mut self, nat: &Nat) {
         match nat {
-            Nat::Ident(ident) => self
-                .set
-                .extend(std::iter::once(IdentKinded::new(ident, Kind::Nat))),
+            Nat::Ident(ident) => {
+                self.set.insert(IdentKinded::new(ident, Kind::Nat));
+            }
```

Comment:
Müssen wir hier (und bei den anderen identifiern) nicht auch noch überprüfen, ob `ident` vielleicht schon in `bound_idents` enthalten ist?
Sonst funktioniert doch auch die Konstruktion in Zeile 100-107 nicht oder?